src/nvim/main.c

```diff
@@ -2070,3 +2072,29 @@ static void check_swap_exists_action(void)
     getout(1);
   handle_swap_exists(NULL);
 }
+
+
+// Check if $VIMRUNTIME points to a valid Neovim runtime path,
+// look for $VIMRUNTIME/autoload/provider/clipboard.vim,
+// which is a Neovim-specific file.
+// Print a warning if the file is not found.
+static void check_vimruntime(void)
+{
+  const char *runtime = os_getenv("VIMRUNTIME");
+  if (runtime) {
+    char rt_testpath[MAXPATHL];
+    STRNCPY(rt_testpath, runtime, MAXPATHL);
+    append_path(rt_testpath, "autoload", MAXPATHL);
+    append_path(rt_testpath, "provider", MAXPATHL);
+    append_path(rt_testpath, "clipboard.vim", MAXPATHL);
+
+    if (!os_file_exists((char_u *)rt_testpath)) {
+      char warning_message[MAXPATHL*2];
+      vim_snprintf(warning_message, sizeof(warning_message),
+          _("W19: The VIMRUNTIME specified (%s) is not a valid Neovim runtime."
+            " %s could not be found.\n"),
```

Comment:
This is enough, but readily available information is better.

By the way, second %s may be just autoload/provider/clipboard.vim. No need to repeat full runtime.